src/solvers/flattening/boolbv_extractbits.cpp

```diff
@@ -24,50 +24,52 @@ bvt boolbvt::convert_extractbits(const extractbits_exprt &expr)
     throw 0;
   }
 
-  mp_integer o1, o2;
-  const bvt &bv0=convert_bv(expr.op0());
+  mp_integer upper_as_int, lower_as_int;
+  auto const &src_bv = convert_bv(expr.src());
 
   // We only do constants for now.
   // Should implement a shift here.
-  if(to_integer(expr.op1(), o1) ||
-     to_integer(expr.op2(), o2))
+  if(
+    to_integer(expr.op1(), upper_as_int) ||
+    to_integer(expr.op2(), lower_as_int))
```

Comment:
While at it: Use a proper API instead of `op1`, `op2`?

Reply:
This is done in a later commit to avoid doing too many things in one commit